### PR TITLE
Updating Last Window Behavior

### PR DESCRIPTION
**Pull request by jleandroperez**
### Details:
In this PR we're restoring the "Dismissing Last Window won't close Simplenote" behavior.

### Testing:
1. Launch Simplenote
2. Click the `x` at the top left corner of the window
3. In the system menu, click over `Note` > `New`
4. Verify that the app is back in the foreground
5. Close Simplenote again
6. Click over the Simplenote icon in the Dock
7. Verify that we're back onscreen!

@roundhill may i bug you with a quick review?
Thanks in advance!!!

cc @drw158 apologies about the late fix, it's been a long road!!

### CODE REVIEW

**Comment by drw158** [comment]
Cool! Thanks!

So does this mean we added an item under the Window menu? That's why the app was rejected by Apple if I'm not mistaken.

**Comment by jleandroperez** [comment]
@drw158 i've opted for your `New note` (ensure Window is Onscreen) suggestion. But on second (or third thoughts), since i'm on this, a Window item doesn't sound _that_ off. Mostly because... you may not wanna create a New Note right?.

Pushing a commit in a couple minutes, thanks Dave!

**Comment by jleandroperez** [comment]
**Update:** Added the `Main Window` menuItem (available via the System Bar's `Menu` item.

Ready for review! thanks!!

**Comment by drw158** [comment]
Main Window makes sense to me

**Comment by roundhill** [comment]
Confirmed both `New Note` and `Main window` options launches a new window 👍

**Comment by roundhill** [comment]
When there was no window showing, I clicked move to trash and the action worked, is that expected behavior?

**Comment by jleandroperez** [comment]
@roundhill niiice catch! i've disabled both, **Print** and **Move to Trash** whenever the main window is not visible.

Ready for another look!

Thank you Dan!!

File: Simplenote/NoteEditorViewController.m [hunk 1]
```diff
@@ -526,11 +526,18 @@ - (void)unpublishNote
...
     // Note menu
+    if (menuItem == deleteItem || menuItem == printItem) {
+        return [[SimplenoteAppDelegate sharedDelegate] isMainWindowVisible];
+    }
     if (menuItem == newItem) {
         return !self.viewingTrash;
-    } else if (menuItem == deleteItem) {
+    }
+
+    if (menuItem == deleteItem) {
```
Comment (roundhill) [inline Simplenote/NoteEditorViewController.m]: This code is never reached now, right? The options are now enabled when I'm viewing the trash.

**Comment by jleandroperez** [comment]
@roundhill my *GOD* what a silly mistake. I totally missed we've got more logic for the `Delete` / `Print` actions below.

Fixed now, thanks!

**Comment by roundhill** [comment]
Looks good! :shipit:

**Comment by jleandroperez** [comment]
Thanks for the review @roundhill!!

**Comment by roundhill** [comment]
> Thanks for the review @roundhill!!

No problem! ❤️

**Pull request merged**